static/js/stream_edit.js

```diff
@@ -219,6 +237,16 @@ function show_subscription_settings(sub_row) {
             return item.email;
         },
     });
+    // To check if all realm users are added
+    // Realm members' emails
+    const active_realm_emails = get_email_of_realm_members(people.get_active_human_persons());
+    // Stream subscribers' emails
+    const filtered_sub_emails = exports.get_email_of_active_subscribers(sub.subscribers);
+    // Displaying "No users to add"
+    if (JSON.stringify(active_realm_emails) === JSON.stringify(filtered_sub_emails)) {
```

Comment:
It seems like there should be a more efficient way to do this, as it seems like a lot of extra work being done.  Not really sure how off the top of my head, but it seems like `if realm_members - subscribers == 0` then there's nobody else to add.

Reply:
I cannot find a solution along this line. The solutions I found either suggest stringify or involve looping over both the arrays. Could you please help me in this a bit more?

Reply:
I believe `people.get_realm_count()`  will get you the number of active users in the realm.  And `static/js/stream_data.js` has various functions that could prove useful.

Reply:
Sorry, but I'm confused here. Are we discussing about the stringify part or the lists?
`people.get_realm count()` return size of active users. `active_realm_emails` is the list of active users (list of emails of people used in `get_realm_count()`). `filtered_sub_emails` returns only people subscribers (non-bots). I am not sure how the no. of users will help us here.

Reply:
Do you want me to compare the list of people objects directly instead of emails?

Reply:
> Sorry, but I'm confused here. Are we discussing about the stringify part or the lists?

Both, I'm curious why you chose to take this approach instead of something like
```js
if (people.get_realm_count() - stream_data.get_subscriber_count(stream_name) === 0) {
    // No more users to add
}
```

The above  and `if (JSON.stringify(active_realm_emails) === JSON.stringify(filtered_sub_emails))` are working to achieve the same goal, no?

> Do you want me to compare the list of people objects directly instead of emails?

My questions is what is the reasoning for aggregating a list of emails and then comparing string representations of those lists as a means to determine if there are no more users to add?  

These are important things to put in a commit message and/or PR description, Ie, explaining why you made the changes, and why they are the correct changes. :)

Reply:
I realized the mistake I was doing. Unfortunately, this did not strike me for a long time. Thanks for pointing this out.
Compiling a list of people/emails is completely unnecessary. I'll proceed with this idea.